projects/core/src/lib/layers/symbology/raster-symbology-editor/raster-symbology-editor.component.ts

```diff
@@ -162,9 +183,23 @@ export class RasterSymbologyEditorComponent implements OnChanges, OnDestroy, Aft
             throw new Error('unsupported colorizer type');
         }
 
+        this.unappliedChanges = true;
+    }
+
+    applyChanges(): void {
+        this.colorMapSelector.applyChanges();
+        this.unappliedChanges = false;
         this.update();
     }
 
+    getNotified(): void {
+        this.unappliedChanges = true;
+    }
+
+    resetChanges(layer: Layer): void {
```

Comment:
I tried at first to implement it like that and preserve the histogram. But I'm not sure if the last emitted value from the histogram observable is still available somewhere. If it is, I couldn't find how/where to access it. If it's no longer accessible I would need to store it somewhere outside of the component each time a new histogram is created... but where would be a good place?

Another way to go about resetting the changes while preserving the histogram might be to store the values/settings for all the other sliders beforehand, and make the reset button fill the previous values back in. That way the component wouldn't be recreated and the histogram isn't touched. But perhaps it could be a bit intrusive to the rest of the code if done like this.

Reply:
```diff
diff --git a/projects/core/src/lib/layers/symbology/raster-symbology-editor/raster-symbology-editor.component.ts b/projects/core/src/lib/layers/symbology/raster-symbology-editor/raster-symbology-editor.component.ts
index 745c495f..37b4927c 100644
--- a/projects/core/src/lib/layers/symbology/raster-symbology-editor/raster-symbology-editor.component.ts
+++ b/projects/core/src/lib/layers/symbology/raster-symbology-editor/raster-symbology-editor.component.ts
@@ -197,7 +197,10 @@ export class RasterSymbologyEditorComponent implements OnChanges, OnDestroy, Aft
     }
 
     resetChanges(layer: Layer): void {
-        this.layoutService.setSidenavContentComponent({component: RasterSymbologyEditorComponent, config: {layer}});
+        this.layoutService.setSidenavContentComponent({
+            component: RasterSymbologyEditorComponent,
+            config: {layer, histogramData: this.histogramData, histogramCreated: this.histogramCreated},
+        });
     }
 
     getNoDataColor(): ColorAttributeInput {
```

Please check if there are no undesired side effects by doing this

Reply:
Thanks! That's a very nice way of keeping the histogram, didn't know this was possible. I tested the functionality and found no side effects.